pallets/gear-debug/src/mock.rs

```diff
@@ -213,9 +213,9 @@ pub fn run_to_block(n: u64, remaining_weight: Option<u64>) {
         Gear::on_initialize(System::block_number());
         let remaining_weight =
             remaining_weight.unwrap_or(pallet_gear::BlockGasLimitOf::<Test>::get());
-        Gear::on_idle(
-            System::block_number(),
-            Weight::from_ref_time(remaining_weight),
-        );
+        pallet_gear::GasAllowanceOf::<Test>::put(remaining_weight);
```

Comment:
where will we update remaining weight in real node?

Reply:
It's set to the block gas limit in the `pallet_gas`'s `on_initialize()` hook - same as before